File: Práctica Grupal 2/practica2-grupo1-Equipo2/src/gestorAplicacion/clasesDeBiblioteca/Copia.py
from .Libro import Libro
from .Prestable import Prestable

class Copia(Libro, Prestable):
    contadorCopia = 0
    def __init__(self, idCopia, libro, ubicacion):
        super().__init__(libro.get_nombre(), 1, libro.get_isbn(), libro.get_autor(), libro.get_año())
        self.disponibleEvento = True  # Por defecto, disponible para eventos
        self.disponibleParticular = True  # Por defecto, disponible para préstamos particulares
        self.ubicacion = ubicacion
        Copia.contadorCopia += 1
        self.idCopia = Copia.contadorCopia

    @staticmethod
    def get_contadorCopia():
        return Copia.contadorCopia
    
    def get_id(self):
        return self.idCopia

    def get_copia_de(self):
        return self.copiaDe

    def is_disponible_evento(self):
        return self.disponibleEvento

    def set_disponible_evento(self, disponibleEvento):
        self.disponibleEvento = disponibleEvento

    def is_disponible_particular(self):
        return self.disponibleParticular

    def set_disponible_particular(self, disponibleParticular):
        self.disponibleParticular = disponibleParticular

    def get_ubicacion(self):
        return self.ubicacion

    def set_ubicacion(self, ubicacion):
        self.ubicacion = ubicacion

    def is_prestado(self):
        return not (self.disponibleEvento and self.disponibleParticular)

    def __str__(self):
        return self.get_nombre()

    def tipo_recurso(self):
        return "Copia"

File: Práctica Grupal 1/src/gestorAplicacion/paquete1/Copia.java
package gestorAplicacion.paquete1;
import java.io.Serializable;

public class Copia extends Libro implements Serializable, Prestable{
	private static final long serialVersionUID = 1L;
    private int idCopia; // Identificador único de la copia
    private Libro copiaDe; // El libro del que es una copia
    private boolean disponibleEvento; // Indica si la copia está disponible para eventos
    private boolean disponibleParticular; // Indica si la copia está disponible para préstamos particulares
    private Biblioteca ubicacion; // La sede a la que está asignada esta copia

    // Constructor de la clase Copia
    public Copia(int idCopia, Libro libro, Biblioteca ubicacion) {
    	super(libro.getNombre(),1,libro.getIsbn(), libro.getAutor(),libro.getAño());
        this.idCopia = idCopia;
        this.disponibleEvento = true; // Por defecto, disponible para eventos
        this.disponibleParticular = true; // Por defecto, disponible para préstamos particulares
        this.ubicacion = ubicacion;
   	
    }
    
    // Método para obtener el identificador único de la copia
    public int getID() {
        return idCopia;
    }

    // Método para obtener el libro del que es una copia
    public Libro getCopiaDe() {
        return copiaDe;
    }

    // Método para verificar si la copia está disponible para eventos
    public boolean isDisponibleEvento() {
        return disponibleEvento;
    }

    // Método para establecer la disponibilidad de la copia para eventos
    public void setDisponibleEvento(boolean disponibleEvento) {
        this.disponibleEvento = disponibleEvento;
    }

    // Método para verificar si la copia está disponible para préstamos particulares
    public boolean isDisponibleParticular() {
        return disponibleParticular;
    }

    // Método para establecer la disponibilidad de la copia para préstamos particulares
    public void setDisponibleParticular(boolean disponibleParticular) {
        this.disponibleParticular = disponibleParticular;
    }

    // Método para obtener la sede a la que está asignada esta copia
    public Biblioteca getUbicacion() {
        return ubicacion;
    }

    // Método para establecer la sede a la que está asignada esta copia
    public void setUbicacion(Biblioteca ubicacion) {
        this.ubicacion = ubicacion;
    }
    public boolean isPrestado() {
    	return (!(disponibleEvento && disponibleParticular));
    }
    
    public String toString() {
    	return this.getNombre();
    }
    
    public String tipoRecurso() {
    	return "Copia";
    }
}
